### ISSUE TITLE
[BUG] FILE__variables ignored during Mastodon configuration

### ISSUE THREAD

**Issue by marcogiorgio** [linuxserver/docker-mastodon#31]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Hi,
I set some environment variables from a file using the FILE__<env_name> notation but their values are ignored by the application. If you provide the ENV directly without the FILE__ notation all works correctly

### Expected Behavior

Application should use the env variables set from files

### Steps To Reproduce

I am using a docker compose file similar to this one:
<details>
  <summary>docker-compose.yaml</summary>
  
  ```yaml
version: '3'
services:
  mastodon:
    image: lscr.io/linuxserver/mastodon
    container_name: mastodon
    networks:
      - internal_network
      - external_network
    ports:
      - 80:80
      - 443:443
    environment:
      - FILE__DB_USER=/run/secrets/postgres_user
      - FILE__DB_NAME=/run/secrets/postgres_db
      - FILE__DB_PASS=/run/secrets/postgres_password
      - FILE__SECRET_KEY_BASE=/run/secrets/secret_key_base
      - FILE__OTP_SECRET=/run/secrets/otp_secret
      - FILE__VAPID_PRIVATE_KEY=/run/secrets/vapid_private_key
      - FILE__VAPID_PUBLIC_KEY=/run/secrets/vapid_public_key
      - FILE__SMTP_PASSWORD=/run/secrets/smtp_password
    volumes:
      - mastodon:/config
    secrets:
      - postgres_password
      - postgres_user
      - postgres_db
      - secret_key_base
      - otp_secret
      - vapid_private_key
      - vapid_public_key
      - smtp_password
    depends_on:
      - db
      - redis
      - es
    labels:
      - swag=enable
    restart: always

volumes:
    mastodon:
  
secrets:
    postgres_password:
      file: ./Secrets/postgres_password.txt
    postgres_user:
      file: ./Secrets/postgres_user.txt
    postgres_db:
      file: ./Secrets/postgres_db.txt
    secret_key_base:
      file: ./Secrets/secret_key_base.txt
    otp_secret:
      file: ./Secrets/otp_secret.txt
    vapid_private_key:
      file: ./Secrets/vapid_private_key.txt
    vapid_public_key:
      file: ./Secrets/vapid_public_key.txt
    smtp_password:
      file: ./Secrets/smtp_password.txt

networks:
  internal_network:
    internal: true
  external_network:

  ```
  
</details>

If you log into the container after the creation and try to run this command you get the errors shown in the log:
`tootctl accounts modify user --role Owner
`
The only FILE__variable that works seems to be `FILE__SMTP_PASSWORD` because I was able to send emails from Mastodon

### Environment

```markdown
- OS:
- How docker service was installed:
```

### CPU architecture

x86-64

### Docker creation

```bash
sudo docker-compose up -d
```

### Container logs

<details>
  <summary>Error when DB_PASS was set from file</summary>
  
```bash  
/app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/connection_adapters/postgresql_adapter.rb:83:in `rescue in new_client': connection to server at "172.19.0.3", port 5432 failed: fe_sendauth: no password supplied (ActiveRecord::ConnectionNotEstablished)
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/connection_adapters/postgresql_adapter.rb:77:in `new_client'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/connection_adapters/postgresql_adapter.rb:37:in `postgresql_connection'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/connection_adapters/abstract/connection_pool.rb:882:in `public_send'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/connection_adapters/abstract/connection_pool.rb:882:in `new_connection'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/connection_adapters/abstract/connection_pool.rb:926:in `checkout_new_connection'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/connection_adapters/abstract/connection_pool.rb:905:in `try_to_checkout_new_connection'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/connection_adapters/abstract/connection_pool.rb:866:in `acquire_connection'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/connection_adapters/abstract/connection_pool.rb:588:in `checkout'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/connection_adapters/abstract/connection_pool.rb:428:in `connection'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/connection_adapters/abstract/connection_pool.rb:1128:in `retrieve_connection'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/connection_handling.rb:327:in `retrieve_connection'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/connection_handling.rb:283:in `connection'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/model_schema.rb:553:in `load_schema!'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/attributes.rb:250:in `load_schema!'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/model_schema.rb:539:in `block in load_schema'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/model_schema.rb:536:in `synchronize'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/model_schema.rb:536:in `load_schema'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/model_schema.rb:402:in `attribute_types'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/model_schema.rb:428:in `type_for_attribute'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/type_caster/map.rb:16:in `type_for_attribute'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/arel/table.rb:108:in `type_for_attribute'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/table_metadata.rb:18:in `type'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation/predicate_builder.rb:63:in `build'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation/predicate_builder.rb:58:in `[]'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation/predicate_builder.rb:131:in `block in expand_from_hash'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation/predicate_builder.rb:84:in `each'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation/predicate_builder.rb:84:in `flat_map'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation/predicate_builder.rb:84:in `expand_from_hash'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation/predicate_builder.rb:29:in `build_from_hash'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation/query_methods.rb:1091:in `build_where_clause'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation/query_methods.rb:46:in `not'
	from /app/www/app/models/account.rb:97:in `block in <class:Account>'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation.rb:411:in `instance_exec'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation.rb:411:in `block in _exec_scope'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation.rb:804:in `_scoping'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation.rb:411:in `_exec_scope'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/scoping/named.rb:176:in `block in scope'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation/delegation.rb:67:in `block in remote'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation.rb:406:in `block in scoping'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation.rb:804:in `_scoping'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation.rb:406:in `scoping'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation/delegation.rb:67:in `remote'
	from /app/www/app/models/account.rb:113:in `block in <class:Account>'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation.rb:411:in `instance_exec'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation.rb:411:in `block in _exec_scope'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation.rb:804:in `_scoping'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation.rb:411:in `_exec_scope'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/scoping/named.rb:176:in `block in scope'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation/delegation.rb:67:in `block in without_unapproved'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation.rb:406:in `block in scoping'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation.rb:804:in `_scoping'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation.rb:406:in `scoping'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation/delegation.rb:67:in `without_unapproved'
	from /app/www/app/models/account.rb:114:in `block in <class:Account>'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation.rb:411:in `instance_exec'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation.rb:411:in `block in _exec_scope'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation.rb:804:in `_scoping'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/relation.rb:411:in `_exec_scope'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activerecord-6.1.7/lib/active_record/scoping/named.rb:176:in `block in scope'
	from /app/www/app/chewy/accounts_index.rb:26:in `<class:AccountsIndex>'
	from /app/www/app/chewy/accounts_index.rb:3:in `<top (required)>'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/bootsnap-1.13.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/bootsnap-1.13.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/kernel.rb:27:in `require'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader/helpers.rb:127:in `const_get'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader/helpers.rb:127:in `cget'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader.rb:239:in `block (2 levels) in eager_load'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader/helpers.rb:41:in `block in ls'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader/helpers.rb:27:in `each'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader/helpers.rb:27:in `ls'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader.rb:234:in `block in eager_load'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader.rb:219:in `synchronize'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader.rb:219:in `eager_load'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader.rb:318:in `each'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/loader.rb:318:in `eager_load_all'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/application/finisher.rb:133:in `block in <module:Finisher>'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/initializable.rb:32:in `instance_exec'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/initializable.rb:32:in `run'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/initializable.rb:61:in `block in run_initializers'
	from /usr/lib/ruby/3.0.0/tsort.rb:228:in `block in tsort_each'
	from /usr/lib/ruby/3.0.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
	from /usr/lib/ruby/3.0.0/tsort.rb:431:in `each_strongly_connected_component_from'
	from /usr/lib/ruby/3.0.0/tsort.rb:349:in `block in each_strongly_connected_component'
	from /usr/lib/ruby/3.0.0/tsort.rb:347:in `each'
	from /usr/lib/ruby/3.0.0/tsort.rb:347:in `call'
	from /usr/lib/ruby/3.0.0/tsort.rb:347:in `each_strongly_connected_component'
	from /usr/lib/ruby/3.0.0/tsort.rb:226:in `tsort_each'
	from /usr/lib/ruby/3.0.0/tsort.rb:205:in `tsort_each'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/initializable.rb:60:in `run_initializers'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/application.rb:391:in `initialize!'
	from /app/www/config/environment.rb:5:in `<top (required)>'
	from /app/www/lib/mastodon/media_cli.rb:4:in `require_relative'
	from /app/www/lib/mastodon/media_cli.rb:4:in `<top (required)>'
	from /app/www/lib/cli.rb:4:in `require_relative'
	from /app/www/lib/cli.rb:4:in `<top (required)>'
	from /app/www/bin/tootctl:5:in `require_relative'
	from /app/www/bin/tootctl:5:in `<main>'
```
</details>

<details>
  <summary>Error when SECRET_KEY_BASE was set from file</summary>
  
```bash
/app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/application.rb:603:in `validate_secret_key_base': Missing `secret_key_base` for 'production' environment, set this string with `bin/rails credentials:edit` (ArgumentError)
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/application.rb:437:in `secret_key_base'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/devise-4.8.1/lib/devise/secret_key_finder.rb:24:in `key_exists?'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/devise-4.8.1/lib/devise/secret_key_finder.rb:16:in `find'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/devise-4.8.1/lib/devise/rails.rb:37:in `block in <class:Engine>'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/initializable.rb:32:in `instance_exec'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/initializable.rb:32:in `run'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/initializable.rb:61:in `block in run_initializers'
	from /usr/lib/ruby/3.0.0/tsort.rb:228:in `block in tsort_each'
	from /usr/lib/ruby/3.0.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
	from /usr/lib/ruby/3.0.0/tsort.rb:431:in `each_strongly_connected_component_from'
	from /usr/lib/ruby/3.0.0/tsort.rb:349:in `block in each_strongly_connected_component'
	from /usr/lib/ruby/3.0.0/tsort.rb:347:in `each'
	from /usr/lib/ruby/3.0.0/tsort.rb:347:in `call'
	from /usr/lib/ruby/3.0.0/tsort.rb:347:in `each_strongly_connected_component'
	from /usr/lib/ruby/3.0.0/tsort.rb:226:in `tsort_each'
	from /usr/lib/ruby/3.0.0/tsort.rb:205:in `tsort_each'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/initializable.rb:60:in `run_initializers'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/application.rb:391:in `initialize!'
	from /app/www/config/environment.rb:5:in `<top (required)>'
	from /app/www/lib/mastodon/media_cli.rb:4:in `require_relative'
	from /app/www/lib/mastodon/media_cli.rb:4:in `<top (required)>'
	from /app/www/lib/cli.rb:4:in `require_relative'
	from /app/www/lib/cli.rb:4:in `<top (required)>'
	from /app/www/bin/tootctl:5:in `require_relative'
	from /app/www/bin/tootctl:5:in `<main>'
  ```
</details>

<details>
  <summary>Error when OTP_SECRET was set from file</summary>
  
```bash
/app/www/config/environments/production.rb:143:in `fetch': key not found: "OTP_SECRET" (KeyError)
	from /app/www/config/environments/production.rb:143:in `block in <top (required)>'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/railtie.rb:234:in `instance_eval'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/railtie.rb:234:in `configure'
	from /app/www/config/environments/production.rb:1:in `<top (required)>'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/bootsnap-1.13.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/bootsnap-1.13.0/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:32:in `require'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/zeitwerk-2.6.0/lib/zeitwerk/kernel.rb:35:in `require'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.7/lib/active_support/dependencies.rb:332:in `block in require'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.7/lib/active_support/dependencies.rb:299:in `load_dependency'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/activesupport-6.1.7/lib/active_support/dependencies.rb:332:in `require'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/engine.rb:571:in `block (2 levels) in <class:Engine>'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/engine.rb:570:in `each'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/engine.rb:570:in `block in <class:Engine>'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/initializable.rb:32:in `instance_exec'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/initializable.rb:32:in `run'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/initializable.rb:61:in `block in run_initializers'
	from /usr/lib/ruby/3.0.0/tsort.rb:228:in `block in tsort_each'
	from /usr/lib/ruby/3.0.0/tsort.rb:350:in `block (2 levels) in each_strongly_connected_component'
	from /usr/lib/ruby/3.0.0/tsort.rb:422:in `block (2 levels) in each_strongly_connected_component_from'
	from /usr/lib/ruby/3.0.0/tsort.rb:431:in `each_strongly_connected_component_from'
	from /usr/lib/ruby/3.0.0/tsort.rb:421:in `block in each_strongly_connected_component_from'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/initializable.rb:50:in `each'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/initializable.rb:50:in `tsort_each_child'
	from /usr/lib/ruby/3.0.0/tsort.rb:415:in `call'
	from /usr/lib/ruby/3.0.0/tsort.rb:415:in `each_strongly_connected_component_from'
	from /usr/lib/ruby/3.0.0/tsort.rb:349:in `block in each_strongly_connected_component'
	from /usr/lib/ruby/3.0.0/tsort.rb:347:in `each'
	from /usr/lib/ruby/3.0.0/tsort.rb:347:in `call'
	from /usr/lib/ruby/3.0.0/tsort.rb:347:in `each_strongly_connected_component'
	from /usr/lib/ruby/3.0.0/tsort.rb:226:in `tsort_each'
	from /usr/lib/ruby/3.0.0/tsort.rb:205:in `tsort_each'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/initializable.rb:60:in `run_initializers'
	from /app/www/vendor/bundle/ruby/3.0.0/gems/railties-6.1.7/lib/rails/application.rb:391:in `initialize!'
	from /app/www/config/environment.rb:5:in `<top (required)>'
	from /app/www/lib/mastodon/media_cli.rb:4:in `require_relative'
	from /app/www/lib/mastodon/media_cli.rb:4:in `<top (required)>'
	from /app/www/lib/cli.rb:4:in `require_relative'
	from /app/www/lib/cli.rb:4:in `<top (required)>'
	from /app/www/bin/tootctl:5:in `require_relative'
	from /app/www/bin/tootctl:5:in `<main>'
  ```
  
</details>

**Comment by thespad** [comment]
Your secret files probably haven't been created correctly and are being misread.

What is the output of `cat -e ./Secrets/postgres_user.txt`?

**Comment by marcogiorgio** [comment]
> cat -e ./Secrets/postgres_user.txt

Here it is:
![image](https://user-images.githubusercontent.com/78936558/214548917-19d1237a-39a2-4223-a406-75890a44cf0a.png)

**Comment by thespad** [comment]
And is that output the same (not the content, the format) for the other problematic envs (DB_PASS, etc.), specifically, no line terminator/blank trailing line?

**Comment by marcogiorgio** [comment]
> And is that output the same (not the content, the format) for the other problematic envs (DB_PASS, etc.), specifically, no line terminator/blank trailing line?

Yes I manually checked all the files because I had this problem in the past for another container (I used `nano` without `-L` argument)

**Comment by thespad** [comment]
Hmm, that's odd then. The OTP/SECRET keys at least are alphanumeric only so I can't even see any weirdness with unescaped special characters.

I'll do some tests and see if I can replicate - the logic isn't unique to this container, it's consistent across all of them.

**Comment by marcogiorgio** [comment]
> Hmm, that's odd then. The OTP/SECRET keys at least are alphanumeric only so I can't even see any weirdness with unescaped special characters.
> 
> I'll do some tests and see if I can replicate - the logic isn't unique to this container, it's consistent across all of them.

Thanks, the strange thing is that FILE__SMTP_PASSWORD seems to work, if there is anything I can do to test this more let me know

**Comment by thespad** [comment]
So I've just done the following:
Added
```
services:
    mastodon:
      environment:
        - FILE__DB_PASS=/run/secrets/postgres_password
      secrets:
        - postgres_password

secrets:
  postgres_password:
    file: ./secret/dbpass
```
To my compose (commented DB_PASS out of my env file) and created my secret:
![image](https://user-images.githubusercontent.com/8425502/214552896-d2f4e0d0-5bed-4ae6-9eec-b71c53810e86.png)

And it's started up quite happily, so there doesn't seem to be anything fundamentally broken with how it handles secrets and the FILE__ envs.

**Comment by marcogiorgio** [comment]
I tried to change again the DB_PASS to use the FILE__DB_PASS and the container indeed managed to start successfully. But can you try to do a `sudo docker exec -it mastodon /bin/bash` and then a `tootctl accounts modify user --role Owner`? I got the error again. Is it possible that these commands read the env variables from somewhere else?

**Comment by thespad** [comment]
Ah, yeah.

So s6 sandboxes the container environment for its services before the init runs so when we handle the FILE__ envs we inject them into the s6 environment but *not* the bash shell environment.

Compare:
`printenv | grep DB_PASS`
and
`ls -la /run/s6/container_environment/ | grep "DB_PASS"`

`tootctl` will be reading from the shell environment, which will then be missing the envs.

Need to have a think about how and whether to handle this at the container level or change it on the bases for all containers.

**Comment by marcogiorgio** [comment]
Cool, I am glad that you found the root cause. I think in the past I had the same problem for another container (I don't remember what exactly), but I am pretty sure I ended up using the classic ENV, so it will probably be nicer to fix the issue for all containers :smile:

**Comment by thespad** [comment]
I've been all around this for several hours, and the short answer is there's no way to "backport" envs into the container environment post-creation in a way that's accessible to all shells.

So, I've created aliases for the secret/vapid generation and tootctl which use the s6 contenv directory which we *can* modify at runtime.

**Comment by thespad** [comment]
Once the PR build is complete, you can test it out and make sure it works with your setup.

**Comment by thespad** [comment]
PR image is `lspipepr/mastodon:v4.0.2-pkg-442959b6-pr-33` if you want to test it out.

Updated readme with the alias info is here: https://github.com/linuxserver/docker-mastodon/blob/aliases-main/README.md#application-setup

**Comment by marcogiorgio** [comment]
Thanks for your work! I just tried the image you provided but it seems that I still get the error `/app/www/config/environments/production.rb:143:in `fetch': key not found: "OTP_SECRET" (KeyError)`. If you try like yesterday do you get the same result?

**Comment by thespad** [comment]
Is that running /app/www/bin/tootctl or using the new alias?

**Comment by thespad** [comment]
Running with the PR image and secrets as per https://github.com/linuxserver/docker-mastodon/issues/31#issuecomment-1403472076
![image](https://user-images.githubusercontent.com/8425502/214912757-bd351008-58c9-491a-8b3c-fcdb4b6c054a.png)

**Comment by marcogiorgio** [comment]
Ok, I was about to post a few screenshots. Yeah I edited all the following variables with FILE__, so: 

DB_USER
DB_NAME
DB_PASS
SECRET_KEY_BASE
OTP_SECRET
VAPID_PRIVATE_KEY
VAPID_PUBLIC_KEY

I don't know how many will be involved when running a tootctl command but for sure the OPT, SECRET_KEY and DB_PASS

**Comment by thespad** [comment]
I'm an idiot.

Because the alias is called `tootctl` and it's in `/` it's being superseded by the real `tootctl` in the PATH

Try `docker exec -it mastodon /tootctl version`

**Comment by marcogiorgio** [comment]
Yes it seems to work now! 😄
